.github/workflows/design-system-ci.yml

```diff
@@ -54,4 +64,102 @@ jobs:
               run: yarn test:ui:ci --filter @code-dot-org/component-library --filter @code-dot-org/design-system-storybook
               working-directory: ./frontend
 
+            - name: Upload static files as pages artifact
+              if: github.event_name == 'push'
+              id: deployment
+              uses: actions/upload-pages-artifact@v3
+              with:
+                path: frontend/apps/design-system-storybook/dist
+                name: component-library-storybook
```

Comment:
IMO we're mixing CI and CD here and that makes me nervous - especially in an open repo where we may choose to allow external contributors to open pull requests. Build & Test require different permissions than Deploy, and I'd like to keep those separate. Is this bundled together mostly to avoid duplication across two workflow files?

Reply:
Yes, I believe @snickell had mentioned this is a preference to avoid an explosion of workflows but I do see a security posture where we separate PRs from deployment scripts. Would you prefer a separate workflow for long term maintenace?

Reply:
Indeed, it is [possible](https://github.com/code-dot-org/code-dot-org/actions/runs/13336596279/job/37253329106?pr=63616) to remove the push conditionals resulting in a potential deploy from a PR. Going to refactor this.

Reply:
As it turns out, modifying a workflow in a PR only executes if you've previously contributed. this can be seen in #63987. 

I did end up refactoring the workflows to have one for PRs and another for push. The build & test one is now callable by the push. Deploy is only done via push.